DTS-STN/canadian-dental-care-plan
# Refactoring wsaddress-service to accept specific address fields rather than DTO

### Description
Accepting specific address fields rather than the DTO would mean if the WSAddress fields change, we would only have to change it within the service and not every occurrence of where the service is used. 

Minor updates included:
- Moved `geographicScope` and `parseType` fields from the validate request to parse request (For reference: [parse request ](https://dev.azure.com/DTS-STN/Canada%20Dental%20Care%20Plan/_wiki/wikis/Canada-Dental-Care-Plan.wiki/252/Parse#) and [validate request](https://dev.azure.com/DTS-STN/Canada%20Dental%20Care%20Plan/_wiki/wikis/Canada-Dental-Care-Plan.wiki/256/Validate#))
- Removing all checks in the service where status is `Valid` and returning the entire WSAddress response (for now). We don't want the service to throw an error if the status is not `Valid` for all operations.

### Related Azure Boards Work Items
[AB#2705](https://dev.azure.com/DTS-STN/Canada%20Dental%20Care%20Plan/_workitems/edit/2705)

### Checklist

### Test Instructions
Run the app locally and make an address change. There should be no errors.
Change the `ws-address-api.server.ts` mock so that the validate service returns "Invalid". Make a home address change. You should be shown the "Address Accuracy" page.
